cmd/deploy/deploy.go

```diff
@@ -212,48 +213,42 @@ func (dc *DeployCommand) RunDeploy(ctx context.Context, deployOptions *Options)
 		}
 
 		oktetoLog.Information("Checking to build services at manifest")
-		if deployOptions.Manifest.Build != nil {
-
-			for service, mOptions := range deployOptions.Manifest.Build {
-				buildOptions := build.BuildOptions{OutputMode: oktetoLog.GetOutputFormat()}
-				if mOptions.Name == "" {
-					mOptions.Name = deployOptions.Name
+		if deployOptions.Build {
+			// force build of all services
+			for service, mBuildInfo := range deployOptions.Manifest.Build {
+				if err := runBuildAndSetEnvs(ctx, service, mBuildInfo); err != nil {
+					return err
 				}
-				opts := build.OptsFromManifest(service, mOptions, buildOptions)
+			}
 
-				if deployOptions.Build {
-					oktetoLog.Debug("force build from manifest definition")
-					if err := runBuildAndSetEnvs(ctx, service, opts); err != nil {
-						return err
-					}
-					continue
-				}
+		} else {
+			// check if images are at registry (global or dev) and set envs or send to build
+			toBuild := make(chan string, len(deployOptions.Manifest.Build))
+
+			g, _ := errgroup.WithContext(ctx)
+			for service, mBuildInfo := range deployOptions.Manifest.Build {
+				service, mBuildInfo := service, mBuildInfo
+				g.Go(func() error {
+					return checkServicesToBuild(service, mBuildInfo, toBuild)
+				})
+			}
 
-				if build.ShouldOptimizeBuild(opts.Tag) {
-					oktetoLog.Debug("found OKTETO_GIT_COMMIT, optimizing the build flow")
-					if skipBuild, err := checkImageAtGlobalAndSetEnvs(service, opts); err != nil {
-						return err
-					} else if skipBuild {
-						continue
-					}
-				}
+			if err := g.Wait(); err != nil {
+				return err
+			}
+			close(toBuild)
 
-				if imageWithDigest, err := registry.GetImageTagWithDigest(opts.Tag); err == oktetoErrors.ErrNotFound {
-					oktetoLog.Debugf("image not found, building image %s", opts.Tag)
-					if err := runBuildAndSetEnvs(ctx, service, opts); err != nil {
-						return err
-					}
-					continue
-				} else if err != nil {
-					return fmt.Errorf("error checking image at registry %s: %v", opts.Tag, err)
-				} else {
-					oktetoLog.Success("Skipping build for image for service %s found: %s", service, opts.Tag)
-					if err := setManifestEnvVars(service, imageWithDigest); err != nil {
-						return err
-					}
+			for {
+				svc, ok := <-toBuild
```

Comment:
I think you can iterate directly over the channel as you don't need a `select` statement and you are closing the channel. I think you could do:

```
for svc := range toBuild {
}
```

Reply:
i saw the first comment first... haha i think this is a better approach, so i will change the code for this option